packages/kmod-5.10-nvidia/kmod-5.10-nvidia.spec

```diff
@@ -0,0 +1,200 @@
+%global nvidia_tesla_470_version 470.57.02
+
+Name: %{_cross_os}kmod-5.10-nvidia
+Version: 1.0.0
+Release: 1%{?dist}
+Summary: NVIDIA drivers for the 5.10 kernel
+# We use these licences because we only ship our own software in the main package,
+# each subpackage includes the LICENSE file provided by the run scripts
+License: Apache-2.0 OR MIT
+URL: http://www.nvidia.com/
+
+# NVIDIA .run scripts from 0 to 199
+Source0: https://us.download.nvidia.com/tesla/%{nvidia_version}/NVIDIA-Linux-%{_cross_arch}-%{nvidia_tesla_470_version}.run
+
+# Common NVIDIA conf files from 200 to 299
+Source200: nvidia-tmpfiles.conf.in
+Source202: nvidia-dependencies-modules-load.conf
+
+# NVIDIA tesla conf files from 300 to 399
+Source300: nvidia-tesla-tmpfiles.conf.in
+Source301: nvidia-tesla-build-config.toml.in
+Source302: nvidia-tesla-path.env.in
+Source303: nvidia-ld.so.conf.in
+
+BuildRequires: %{_cross_os}glibc-devel
+BuildRequires: %{_cross_os}kernel-5.10-archive
+Requires: %{_cross_os}binutils
+
+%description
+%{summary}.
+
+%package tesla-470
+Summary: NVIDIA 470 Tesla driver
+Version: %{nvidia_tesla_470_version}
+License: LicenseRef-NVIDIA-Tesla-driver-Linux
+Requires: %{name}
+
+%description tesla-470
+%{summary}
+
+%prep
+# Extract nvidia sources with `-x`, otherwise the script will try to install
+# the driver in the current run
+sh %{S:0} -x
+
+%global kernel_sources %{_builddir}/kernel-devel
+tar -xf %{_cross_datadir}/bottlerocket/kernel-devel.tar.xz
+
+%build
+pushd NVIDIA-Linux-%{_cross_arch}-%{nvidia_tesla_470_version}/kernel
+
+# This recipe was based in the NVIDIA yum/dnf specs:
+# https://github.com/NVIDIA/yum-packaging-precompiled-kmod
+
+# We set IGNORE_CC_MISMATCH even though we are using the same compiler used to compile the kernel, if
+# we don't set this flag the compilation fails
+make %{?_smp_mflags} IGNORE_CC_MISMATCH=1 SYSSRC=%{kernel_sources} CC=%{_cross_target}-gcc
+
+%{_cross_target}-strip -g --strip-unneeded nvidia/nv-interface.o
+%{_cross_target}-strip -g --strip-unneeded nvidia-uvm.o
+%{_cross_target}-strip -g --strip-unneeded nvidia-drm.o
+%{_cross_target}-strip -g --strip-unneeded nvidia-peermem/nvidia-peermem.o
+%{_cross_target}-strip -g --strip-unneeded nvidia-modeset/nv-modeset-interface.o
+
+# We delete these files since we just stripped the input .o files above, and
+# will be build at runtime in the host
+rm nvidia{,-modeset,-peermem}.o
+
+%{_cross_target}-ld -r -o nvidia-modeset.o nvidia-modeset/nv-modeset-interface.o nvidia-modeset/nv-modeset-kernel.o
+%{_cross_target}-ld -r -o nvidia-peermem.o nvidia-peermem/nvidia-peermem.o
```

Comment:
The first `ld` invocation here seems useless, since we don't actually package the resulting file. The second one is creating a relocatable version of `nvidia-peermem.o` which we then install. This seems like a subtle side effect in the dnf spec, and I'm not entirely sure it's required.

In the interests of clarity, I'd remove both these lines.

If we end up needing the `ld -r` invocation for `nvidia-peermem.o`, I'd add an "object file" step so `driverdog` can generate it from `nvidia-peermem/nvidia-peermem.o`, like we do for `nvidia-modeset.o`.